modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/DeleteContentCommand.java

```diff
@@ -144,15 +163,22 @@ public void totalToDelete( final int count )
 
     private NodeIds deleteNodeInDraftAndMaster( final NodeId nodeToDelete )
     {
-        final Context currentContext = ContextAccessor.current();
-        final NodeIds draftNodes = deleteNodeInContext( nodeToDelete, currentContext );
-        final NodeIds masterNodes = deleteNodeInContext( nodeToDelete, ContextBuilder.from( currentContext ).
+        final Context draftContext = ContextAccessor.current();
+        final Context masterContext = ContextBuilder.from( draftContext ).
             branch( ContentConstants.BRANCH_MASTER ).
-            build() );
-        return masterNodes != null ? masterNodes : draftNodes;
+            build();
+
+        final NodeIds draftNodes = deleteNodeInContext( nodeToDelete, draftContext );
+        final NodeIds masterNodes = deleteNodeInContext( nodeToDelete, masterContext );
+
+        draftNodes.stream().
+            filter( id -> !masterNodes.contains( id ) ).
+            forEach( id -> deleteNodeInContext( id, masterContext ) );
+
+        return draftNodes;
```

Comment:
This may be confusing, code deletes nodes from master and draft but reports only draft.
So it is unclear for event/auditlog which nodes were unpunblished.